numpy/_distributor_init.py

```diff
@@ -1,10 +1,12 @@
 """ Distributor init file
 
-Distributors: you can add custom code here to support particular distributions
-of numpy.
+Distributors: you can add a _distributor_init_local.py file to support
```

Comment:
I'd like to revert this doc change, since it's better for distributors to overwrite this file, for two reasons:
1. no need to change existing build scripts
2. keeps `python -m build` working, while adding an untracked `_distributor_init_local.py` file would not work since that file will not be included in the sdist (and wheel is built from sdist)

Reply:
The original reason I made a change at all here is that it was too easy to run the scipy-openblas workflow locally (which modifies this file) and then mistakenly check the modified file in to VCS along with a bunch of other changes. If we revert this, perhaps we should add a check as part of the smoke test that the file has not been modified?

Reply:
No need I think - we always had this file, and it's just not changing. This is no longer used by `scipy-openblas` (that uses `_local`) and for redistributions things work exactly like before.

Reply:
Ahh, got it, just revert the doc change, leave the contents. OK. Will adopt the other suggestions here too.

Reply:
This occurs with submodules and licence files frequently. I wish git had some facility that would complain if you tried to commit to a specified set of files, that you would have to give a force flag to overcome that resistance.

Reply:
Yes, that would be nice. Too many contributors make mistakes with git submodules. And I even do it myself on occasion, since `git commit -a` is kinda muscle memory.

Reply:
We could add a check as part of smoke_test that all the files in a `do_not_change.txt` (including submodules) are not part of a PR. Then disable that check if the PR has the "allow_changed_immutable_state" label. The check would only run on PRs, not on merge to main.

Reply:
Is this link useful, https://superuser.com/questions/1301581/get-git-commit-a-to-ignore-submodules

Reply:
I'm not so sure if for either of those the cure isn't as problematic as the disease. If I had to choose, I think the one from @andyfaff's link is nicer. The downside is:

_If you set ignore = all, to get sane behavior with git commit -a, it will ALSO ignore the submodule in git show/diff when you EXPLICITLY add them. The only way to work-around the latter is using the command line option --ignore-submodule=none._

I'd probably leave it alone in NumPy; in SciPy it seems more like a toss-up.